src/infrastructure/cli.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "CLI11/CLI11.hpp"
+#include "infrastructure/logger.hpp"
+#include "infrastructure/terminator.hpp"
+
+namespace serac {
+
+namespace cli {
+
+//------- Command Line Interface -------
+
+std::shared_ptr<std::unordered_map<std::string, std::string>> defineAndParse(int argc, char* argv[], int rank)
```

Comment:
Do we need shared ownership of this and the Inlet object?  From what I can see, the lifetime of the driver and the lifetime of the inlet object are closely coupled enough that the driver can be responsible for it, in which case both this `unordered_map` and the `inlet` object can be on the stack (declared in `main`, returned by value from these utility functions).

Reply:
I removed the `shared_ptr` from the `datastore` and `cli_opts` but due to some possibly ill-advised decisions in Inlet i couldn't remove those.  We can talk about better ways to handle this tomorrow and implement them in a follow up PR.